python/cudf/cudf/core/accessors/string.py

```
@@ -1064,9 +1073,14 @@ def replace(
             )
         else:
-            result = self._column.replace_str(
+            if case is False:
+                input_col = self._column.to_lower()
+                pat = pat.lower()  # type: ignore[union-attr]
+            else:
+                input_col = self._column
+            result = input_col.replace_str(
                 pat,  # type: ignore[arg-type]
-                repl,
+                pa_repl,
```

Ruff (0.15.14)

[error] Undefined name `pa_repl`

(F821)